action.yml

```diff
@@ -243,6 +245,38 @@ runs:
         CA_GH_ACTION_RESOURCE_GROUP="${{ inputs.containerAppName }}-rg"
         echo "CA_GH_ACTION_RESOURCE_GROUP=${CA_GH_ACTION_RESOURCE_GROUP}" >> $GITHUB_ENV
 
+    - name: Check if the resource group exists
+      shell: bash
+      run: |
+        az group show \
+          -n ${{ env.CA_GH_ACTION_RESOURCE_GROUP }} \
+          -o none && CA_GH_ACTION_RG_EXISTS=true || CA_GH_ACTION_RG_EXISTS=false
+        echo "CA_GH_ACTION_RG_EXISTS=${CA_GH_ACTION_RG_EXISTS}" >> $GITHUB_ENV
+
+    - name: Get default location for Container Apps to use if resource group doesn't exist and location wasn't provided
+      if: ${{ env.CA_GH_ACTION_RG_EXISTS == 'false' && inputs.location == '' }}
+      shell: bash
+      run: |
+        CA_GH_ACTION_DEFAULT_LOCATION=$(az provider show -n Microsoft.App --query "resourceTypes[?resourceType=='containerApps'].locations[] | [0]")
+        CA_GH_ACTION_DEFAULT_LOCATION=$(echo $CA_GH_ACTION_DEFAULT_LOCATION | sed 's/[ !()]//g')
+        CA_GH_ACTION_DEFAULT_LOCATION=${CA_GH_ACTION_DEFAULT_LOCATION,,}
```

Comment:
I'm not familiar with this syntax: could you please confirm that the `,,` is on purpose?

Reply:
I'm glad you commented on this since I wasn't able to make a comment in the YAML file 😄 

For complete transparency about what's going on in this entire block:

- We use `az provider show` to get the locations for the `Microsoft.App/containerApps` resource type and grab the first one
- Strip out any white space and parentheses
- Make the string all lowercase letters

This allows us to take something like "East US 2" and turn it into "eastus2", normalizing the location for future command calls.